src/Microsoft.OData.Core/Parameterized.Microsoft.OData.Core.cs

```diff
@@ -2147,6 +2147,13 @@ internal static string ReaderValidationUtils_NullValueForNonNullableType(object
             return Microsoft.OData.TextRes.GetString(Microsoft.OData.TextRes.ReaderValidationUtils_NullValueForNonNullableType, p0);
         }
 
+        /// <summary>
+        /// A string like "A null value was found with the expected type '{0}[Nullable=True]'. The expected type '{0}[Nullable=True]' does not allow null values."

```

Comment:
>The expected type '{0}[Nullable=True]' does not allow null values. [](start = 95, length = 66)

This message is confusing to me.  Is it used exclusively when a collection-valued property is null? 

Can we make it something like "A null value was found for a collection of type '{0}[Nullable=True]. Collection-valued properties with Nullable=True can contain null values, but collection-valued properties cannot themselves be null."